src/Network/Network/Models/AzureFirewall/PSAzureFirewall.cs

```diff
@@ -386,6 +386,24 @@ public void RemovePublicIpAddress(PSPublicIpAddress publicIpAddress)
             }
         }
 
+        public void AddIpAddressesForByopipHubFirewall(PSPublicIpAddress[] publicIpAddresses)
+        {
+            this.IpConfigurations = new List<PSAzureFirewallIpConfiguration>();
+
+            if (publicIpAddresses != null && publicIpAddresses.Count() > 0)
+            {
+                for (var i = 0; i < publicIpAddresses.Count(); i++)
+                {
+                    this.IpConfigurations.Add(
+                        new PSAzureFirewallIpConfiguration
+                        {
+                            Name = $"{AzureFirewallIpConfigurationName}{i}",
+                            PublicIpAddress = new PSResourceId { Id = publicIpAddresses[i].Id }
```

Comment:
Is null check necessary here?

Reply:
Not necessarily. If you'd see the function from where it is called, I am already doing a null check on the object